### ISSUE TITLE
Buttons with dropdown should feature proper semantic markup

### ISSUE THREAD

**Issue by mlewand** [ckeditor/ckeditor5#12258]
## 📝 Provide a description of the improvement

Extracted from ckeditor/ckeditor5#1037

The screen reader should properly announce that the button has a popup,

We already cover `aria-haspopup` well enough.

However, we must handle `aria-expanded` for dropdown button views. Interestingly [split buttons already have it implemented](https://github.com/ckeditor/ckeditor5/blob/5e8cdcb9e493ddfbff6f56a12cd8599d3aa664b7/packages/ckeditor5-ui/src/dropdown/button/splitbuttonview.js#L227), so this could serve as a reference.

---

If you'd like to see this improvement implemented, add a 👍 reaction to this post.

**Comment by mlewand** [comment]
There has been some discussion in #1037 already regarding this:

[@oleq wrote](https://github.com/ckeditor/ckeditor5/issues/1037#issuecomment-1080599329):

> 1.  Our dropdowns have no `aria-expanded` but should have (see an example from w3c [https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html](https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html))
>     *   The only places that have that attribute are `CollapsibleView` (list properties: start index, reversed section) and `SplitButtonView` (a button for dropdowns with an “extra arrow”).

[@Comandeer wrote](https://github.com/ckeditor/ckeditor5/issues/1037#issuecomment-1147652455):

> > The screen reader should properly announce that the button has a popup,
> 
> It can be done by adding the [`[aria-haspopup]`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup) and [`[aria-expanded]`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded) attributes to buttons triggering dropdowns. Depending on what the dropdown is, the `[aria-haspopup]` attribute can take different values instead of generic `true` (e.g. `menu` if the dropdown is some kind of a menu). We also need to ensure that the dropdown itself has an appropriate role (e.g. `[role=menu]` in case of menu dropdown). ARIA Practices contains a decent [menu button sample](https://www.w3.org/WAI/ARIA/apg/example-index/menu-button/menu-button-actions.html).

**Comment by Comandeer** [comment]
Tested on [[https://ckeditor.com/docs/ckeditor5/latest/examples/builds-custom/full-featured-editor.html](https://ckeditor.com/docs/ckeditor5/latest/examples/builds-custom/full-featured-editor.htmla)](https://ckeditor.com/docs/ckeditor5/latest/examples/builds-custom/full-featured-editor.htmla) and "Text Color" button.

1. The button is correctly marked up as having a submenu by the `[aria-haspopup=true]` attribute

   1.  Optionally: check if we can't find more concrete value for this attribute (this could be tricky as it probably depends on context – some buttons would have menus, some grids, etc.)

2. There is no info about the state of the menu – if it is expanded or collapsed. It could be fixed by adding the `[aria-expanded]` attribute to the button.

3. The menu itself does not have a proper role. It is a [requirement in ARIA spec](https://w3c.github.io/aria/#aria-haspopup):

   > Authors _**MUST**_ ensure that the role of the element that serves as the container for the popup content is [`menu`](https://w3c.github.io/aria/#menu), [`listbox`](https://w3c.github.io/aria/#listbox), [`tree`](https://w3c.github.io/aria/#tree), [`grid`](https://w3c.github.io/aria/#grid), or [`dialog`](https://w3c.github.io/aria/#dialog), and that the value of `aria-haspopup` matches the role of the popup container.

4. We can also think about adding the [`[aria-controls]` attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls) to indicate the relationship between the button and the menu.
